src/main/java/dev/sigstore/json/GsonSupplier.java

```diff
@@ -30,13 +30,23 @@
 public enum GsonSupplier implements Supplier<Gson> {
   GSON;
 
-  private final Gson gson =
-      new GsonBuilder()
-          .registerTypeAdapter(byte[].class, new GsonByteArrayAdapter())
-          .registerTypeAdapterFactory(new GsonAdaptersRekorEntry())
-          .registerTypeAdapterFactory(new GsonAdaptersRekorEntryBody())
-          .disableHtmlEscaping()
-          .create();
+  private final Gson gson;
+
+  GsonSupplier() {
+    GsonBuilder gsonBuilder = new GsonBuilder();
+    for (TypeAdapterFactory factory : ServiceLoader.load(TypeAdapterFactory.class)) {
```

Comment:
does this just autoload all the typeadapters in the classpath?

Reply:
yeah pretty much. Immutables registers them as META-INF/services/com.google.gson.TypeAdapterFactory, and then this code just loads them all up.

Reply:
https://immutables.github.io/json.html

Reply:
I've changed the code back to using explicit registration.